server/server.go

```diff
@@ -0,0 +1,214 @@
+package server
+
+import (
+	"context"
+	"net"
+
+	"net/http"
+
+	"sync"
+
+	"time"
+
+	"errors"
+
+	"github.com/infobloxopen/atlas-app-toolkit/gateway"
+	"github.com/infobloxopen/atlas-app-toolkit/health"
+	"google.golang.org/grpc"
+)
+
+var (
+	ErrInitializeTimeout      = errors.New("initialization timed out")
```

Comment:
This got left out of my earlier review, but these should have comments.

Running `golint` turned this up and a few others:

```
flag.go:11:6: exported type GRPCFlags should have comment or be unexported
flag.go:15:6: exported type TLSFlags should have comment or be unexported
flag.go:21:6: exported type HealthProbesFlags should have comment or be unexported
flag.go:27:1: exported function NewHealthProbesFlags should have comment or be unexported
flag.go:37:1: exported method HealthProbesFlags.Addr should have comment or be unexported
flag.go:41:1: exported method HealthProbesFlags.HealthPath should have comment or be unexported
flag.go:45:1: exported method HealthProbesFlags.ReadyPath should have comment or be unexported
flag.go:49:1: exported function MetricsFlags should have comment or be unexported
flag.go:55:1: exported function NewGRPCFlags should have comment or be unexported
flag.go:61:1: exported method GRPCFlags.Addr should have comment or be unexported
flag.go:65:1: exported function NewTLSFlags should have comment or be unexported
flag.go:74:1: exported method TLSFlags.TLSConfig should have comment or be unexported
flag.go:78:1: exported method TLSFlags.WithGRPCTLSCreds should have comment or be unexported
server.go:21:2: exported var ErrInitializeTimeout should have comment or be unexported
server.go:39:6: exported type Option should have comment or be unexported
server.go:41:6: exported type InitializerFunc should have comment or be unexported
server.go:161:1: exported method Server.Stop should have comment or be unexported
tls.go:1:1: package comment should be of the form "Package server ..."
```

Reply:
good call -- I need to figure out how to get linter errors in my IDE!